backend/app/usecase/shortlink/creator.go

```diff
@@ -64,9 +64,9 @@ type CreatorPersist struct {
 
 // CreateShortLink persists a new short link with a given or auto generated alias in the repository.
 // TODO(issue#235): add functionality for public URLs
-func (c CreatorPersist) CreateShortLink(shortLink entity.ShortLink, customAlias *string, user entity.User, isPublic bool) (entity.ShortLink, error) {
-	longLink := shortLink.LongLink
-	isValid, violation := c.longLinkValidator.IsValid(&longLink)
+func (c CreatorPersist) CreateShortLink(shortLinkInput entity.ShortLinkInput, user entity.User, isPublic bool) (entity.ShortLink, error) {
+	longLink := shortLinkInput.GetLongLink("")
+	isValid, violation := c.longLinkValidator.IsValid(longLink)
```

Comment:
`IsValid` is accepting a pointer. We should pass in the pointer directly.

Reply:
I updated validator interface to accept value instead of pointer, therefore it's ok to pass in value directly. What do you think? Should we change interface back to accepting pointers to check for `nil`?

Reply:
@Coteh Then it's good to go. Let's check whether the alias is `nil` and generate one at the beginning of this function. We generate when alias is not provided.

Reply:
I mean line 84 - line 91

Reply:
Right now, if alias is `nil`, then `customAlias` will be empty string, and if custom alias is empty string then alias will be auto-generated. If I change the logic to what you suggested, then empty string will be a valid alias. Is this something you want?